Assets/HoloToolkit/Input/Scripts/GameControllerManipulator.cs

```diff
@@ -0,0 +1,145 @@
+﻿using System;
+using UnityEngine;
+using System.Collections;
+using HoloToolkit.Unity;
+
+namespace HoloToolkit.Unity
+{
+    /// <summary>
+    /// Allows the user to place and rotate GameObjects using a game controller.
+    /// </summary>
+    /// <remarks>Make sure to enable the HumanInterfaceDevice capability before using.</remarks>
+    public class GameControllerManipulator : MonoBehaviour
+    {
+        [Tooltip("Name of the joystick axis that moves the object along X as set in InputManager")]
+        public string MoveXAxisName = "ControllerLeftStickX";
+        [Tooltip("Speed of movement along X")]
+        public float MoveXAxisSpeed = 0.2f;
+
+        [Tooltip("Name of the joystick axis that moves the object along Y as set in InputManager")]
+        public string MoveYAxisName = "ControllerLeftStickY";
+        [Tooltip("Speed of movement along Y")]
+        public float MoveYAxisSpeed = 0.2f;
+
+        [Tooltip("Name of the joystick axis that moves the object along Z as set in InputManager")]
+        public string MoveZAxisName = "ControllerTriggerAxis";
+        [Tooltip("Speed of movement along Z")]
+        public float MoveZAxisSpeed = 0.2f;
+
+        [Tooltip("Name of the button the user presses to move the object with gaze")]
+        public string MoveWithGazeButtonName = "Fire1";
+
+        [Tooltip("Name of the joystick axis that rotates around X as set in InputManager")]
+        public string RotateAroundXAxisName = "ControllerLeftStickY";
+        [Tooltip("Speed of rotation around X")]
+        public float RotateAroundXAxisSpeed = -1f;
+
+        [Tooltip("Name of the joystick axis that rotates around Y as set in InputManager")]
+        public string RotateAroundYAxisName = "ControllerLeftStickX";
+        [Tooltip("Speed of rotation around Y")]
+        public float RotateAroundYAxisSpeed = -1f;
+
+        [Tooltip("Name of the joystick axis that rotates around Z as set in InputManager")]
+        public string RotateAroundZAxisName = "";
+        [Tooltip("Speed of rotation around Z")]
+        public float RotateAroundZAxisSpeed = 1f;
+
+        [Tooltip("Name of the button used to enable rotation. Useful for single stick controllers")]
+        public string RotateModifierButtonName = "Fire2";
+
+        [Tooltip("Move the gaze target instead of the GameObject this script is attached to")]
+        public bool MoveGazeTarget = false;
+
+        private GameObject lastManipulatedObject;
+        private float stickDistance = 0;
+
+        // Update is called once per frame
+        void Update()
+        {
+            GameObject go;
+
+            if (!MoveGazeTarget)
+                go = gameObject;
```

Comment:
Please consistently use the:
if(condition)
{
    // braces pattern
}
else
{
}